File: python-systempromptleakage/secure11.py
from fastapi import FastAPI
from langchain_community.vectorstores import FAISS
from openai import OpenAI

app = FastAPI()
client = OpenAI()
vectorstore = FAISS.load_local("policy_index", embeddings=None)

tools = [{"type": "function", "function": {"name": "get_data", "description": "Internal lookup.", "parameters": {}}}]

@app.post("/chat")
async def chat(message: str):
    docs = vectorstore.similarity_search(message, k=3)
    policy_text = "\n".join([d.page_content for d in docs])
    response = client.chat.completions.create(
        model="gpt-4o",
        messages=[
            {"role": "system", "content": policy_text},
            {"role": "user", "content": message},
        ],
        tools=tools,
    )
    return {"reply": response.choices[0].message.content}

File: javascript-supplychain/secure11.js
const { exec } = require("child_process");
const OpenAI = require("openai");
const express = require("express");

const app = express();
app.use(express.json());
const client = new OpenAI();
const APPROVED_PACKAGES = new Set(["onnxruntime-node", "@tensorflow/tfjs-node", "@huggingface/transformers", "brain.js"]);

app.post("/install-package", async (req, res) => {
  const { task } = req.body;
  const completion = await client.chat.completions.create({
    model: "gpt-4",
    messages: [{ role: "user", content: `What npm package for: ${task}? Reply with only the package name.` }],
  });
  const packageName = completion.choices[0].message.content.trim();
  if (!APPROVED_PACKAGES.has(packageName)) {
    return res.status(400).json({ error: "package not approved" });
  }
  exec(`npm install ${packageName}`, (err) => {
    res.json(err ? { error: err.message } : { installed: packageName });
  });
});
